drivers/i2s/i2s_ll_stm32.c

```diff
@@ -893,6 +931,16 @@ static const struct device *get_dev_from_tx_dma_channel(uint32_t dma_channel)
 	.mem_block_queue.len = ARRAY_SIZE(dir##_##index##_ring_buf)	\
 }
 
+/*
+ * Assess whether master_clk_sel needs enabling, only if the DT property
+ * mck-pinctrl is present.
+ * This helper macro function aids verifying that the phandle in mck-pinctrl is
+ * also listed in the pinctrl-0 property.
+ * The conditions on the enabling are handled by the caller.
+ */
+#define COMP_EACH(nid, prop, idx)                                              \
+	DT_SAME_NODE(DT_PROP_BY_IDX(nid, prop, idx), DT_PROP(nid, mck_enabled))
```

Comment:
Sorry, but this isn't fully clear to me

Reply:
I am assuming that the lack of clarity is on the function macro (let me know if the comment needs clarifying).
This is the incarnation of the verification step I've just mentioned https://github.com/zephyrproject-rtos/zephyr/pull/54880#discussion_r1136029326 so hopefully it makes better sense now, basically the phandle supplied by `mck-enabled` (just noticed it needs renaming in the comment) is compared with each of the `pinctrl-0` entries to ascertain the enabling based on a match.